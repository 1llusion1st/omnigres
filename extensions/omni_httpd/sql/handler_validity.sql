-- Invalid queries
INSERT INTO omni_httpd.handlers (query) VALUES($$SELECT * FROM no_such_table$$);
INSERT INTO omni_httpd.handlers (query) VALUES($$SELECT request.pth FROM request$$);

-- Valid query at the end of the transaction
BEGIN;
INSERT INTO omni_httpd.handlers (query) VALUES($$SELECT * FROM no_such_table$$);
CREATE TABLE no_such_table ();
END;

DROP TABLE no_such_table;
