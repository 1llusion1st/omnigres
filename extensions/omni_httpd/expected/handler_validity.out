-- Invalid queries
INSERT INTO omni_httpd.handlers (query) VALUES($$SELECT * FROM no_such_table$$);
ERROR:  invalid query
DETAIL:  relation "no_such_table" does not exist
INSERT INTO omni_httpd.handlers (query) VALUES($$SELECT request.pth FROM request$$);
ERROR:  invalid query
DETAIL:  column request.pth does not exist
INSERT INTO omni_httpd.handlers (query) VALUES($$$$);
ERROR:  query can only contain one statement
INSERT INTO omni_httpd.handlers (query) VALUES($$SELECT; SELECT$$);
ERROR:  query can only contain one statement
-- Valid query at the end of the transaction
BEGIN;
INSERT INTO omni_httpd.handlers (query) VALUES($$SELECT * FROM no_such_table$$);
CREATE TABLE no_such_table ();
END;
DROP TABLE no_such_table;
